lib/_stream_readable.js

```diff
@@ -68,11 +68,19 @@ function ReadableState(options, stream) {
   // the point at which it stops calling _read() to fill the buffer
   // Note: 0 is a valid value, means "don't call _read preemptively ever"
   var hwm = options.highWaterMark;
-  var defaultHwm = this.objectMode ? 16 : 16 * 1024;
-  this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
+  if (hwm != null) {
```

Comment:
I did not want to change the behavior of accepting `0`. I just deleted `defaultHwm` because I think it is pretty obvious what `16 * 1024` is in the new code, though I can use a variable as well.